src/msg/async/AsyncConnection.cc

```diff
@@ -859,6 +859,18 @@ ssize_t AsyncConnection::_process_connection()
           cs.close();
         }
 
+        {
+          entity_addr_t addr2bind = msgr->get_myaddr();
+          if (msgr->cct->ms_bind_before_connect && (!addr2bind.is_blank_ip())) {
```

Comment:
get_myaddr() returns by reference, right?  can use it directly instead of copying the value to addr2bind